textdb/textdb-dataflow/src/test/java/edu/uci/ics/textdb/dataflow/keywordmatch/KeywordMatcherTest.java

```diff
@@ -0,0 +1,402 @@
+     *
+     * BooleanQuery() is deprecated. In future a better solution could be worked out in Query builder layer
+     *
+     * @param query
+     * @param attributeList
+     * @return Query
+     * @throws ParseException
+     */
+    private Query createQueryObject(String query, ArrayList<Attribute> attributeList) throws ParseException {
+        Analyzer analyzer = new StandardAnalyzer();
+        ArrayList<String> tokens;
+        String[] fields = new String[attributeList.size()];
+        for (int i=0;i< attributeList.size();i++){
+            fields[i] = attributeList.get(i).getFieldName();
+        }
+
+        tokens = tokenizeQuery(analyzer, query);
+        BooleanQuery booleanQuery = new BooleanQuery();
+        MultiFieldQueryParser parser = new MultiFieldQueryParser(fields, analyzer);
+        for(String searchToken: tokens){
+            Query termQuery = parser.parse(searchToken);
+            booleanQuery.add(termQuery, BooleanClause.Occur.MUST);
+        }
+        return booleanQuery;
+    }
+
+
+    /**
+     * For a given string query & list of attributes it gets a list of results
+     * buildMultiQueryOnAttributeList flag decides if the query is formed as a boolean Query on all attribute
+     * or all records are scanned
+     * @param query
+     * @param attributeList
+     * @param buildMultiQueryOnAttributeList
+     * @return List<ITuple>
+     * @throws DataFlowException
+     * @throws ParseException
+     */
+
+    public List<ITuple> getQueryResults(String query, ArrayList<Attribute> attributeList, boolean buildMultiQueryOnAttributeList) throws DataFlowException, ParseException {
+
+        Analyzer analyzer = new StandardAnalyzer();
+        IPredicate predicate = new KeywordPredicate(query, attributeList, analyzer);
+        QueryParser queryParser;
+        if(!buildMultiQueryOnAttributeList) {
+            queryParser = new QueryParser(TestConstants.ATTRIBUTES_PEOPLE[0].getFieldName(), analyzer);
+            queryObj = queryParser.parse(LuceneConstants.SCAN_QUERY);
+        }
+        else {
+            queryObj = createQueryObject(query, attributeList);
+        }
+        IDataReader dataReader = new LuceneDataReader(dataStore, queryObj);
+        indexSearchSourceOperator = new IndexSearchSourceOperator(dataReader);
+        keywordMatcher = new KeywordMatcher(predicate, indexSearchSourceOperator);
+        keywordMatcher.open();
+        List<ITuple> results = new ArrayList<>();
+        ITuple nextTuple = null;
+        while ((nextTuple = keywordMatcher.getNextTuple()) != null) {
+            results.add(nextTuple);
+        }
+        indexSearchSourceOperator.close();
+        return results;
+    }
+
+    /**
+     * For a given string query it gets a list of results
+     *
+     * @param query
+     * @return List<ITuple>
+     * @throws DataFlowException
+     * @throws ParseException
+     */
+    public List<ITuple> getQueryResults(String query) throws DataFlowException, ParseException {
```

Comment:
Can we use the code of "getQueryResults(String query, ArrayList<Attribute> attributeList, boolean buildMultiQueryOnAttributeList)" defined above by passing the right parameters?